### PR TITLE
Issue 175 make query chap

### PR DESCRIPTION
**Pull request by dsolt**
This PR moves functions related to querying the system in general to their own chapter. This leaves the chapter they were taken from very small, especially when accounting for #390 which also removes material from the same chapter.  However,
yet another PR is forthcoming which will address what remains of the Chap_API_Sync_Access chapter.

### CODE REVIEW

**Comment by dsolt** [comment]
This PR uses 2 commits in a slightly strange way that allows you to look at just the changes for the 2nd commit and see how the material in the new chapter changed from what it was before.   That is hard to see if you just look at the combined changes.

**Comment by dsolt** [comment]
Please use emoji reactions ON THIS COMMENT to indicate your position on this proposal.

    You do not need to vote on every proposal
    If you have no opinion, don't vote - that is also useful data
    If you've already commented on this issue, please still vote so
    we know your current thoughts
    Not all proposals solve exactly the same problem, so we may end
    up accepting proposals that appear to have some overlap
    This is not a binding majority-rule vote, but it will be a very
    significant input into the corresponding ASC decision.

Here are the meanings for the emojis:

    Hooray or Rocket: I support this so strongly that I
    want to be an advocate for it
    Heart: I think this is an ideal solution
    Thumbs up: I'd be happy with this solution
    Confused: I'd rather we not do this, but I can tolerate it
    Thumbs down: I'd be actively unhappy, and may even consider
    other technologies instead
    If you want to explain in more detail, feel free to add another
    comment, but please also vote on this comment.

File: Chap_API_Query.tex [hunk 1]
```diff
@@ -611,21 +211,21 @@ \subsection{\code{PMIx_Query_info_nb}}
...
-
-\ac{PMIx} libraries are not required to directly support any other attributes for this function. However, all provided attributes must be passed to the host \ac{SMS} daemon for processing. The \ac{PMIx} library is \textit{required} to add the \refAttributeItem{PMIX_USERID} and the \refAttributeItem{PMIX_GRPID} attributes of the client process making the request.
+Note that inclusion of both the \refattr{PMIX_PROCID} directive and either the \refattr{PMIX_NSPACE} or the \refattr{PMIX_RANK} attribute will return a \refconst{PMIX_ERR_BAD_PARAM} result, and that the inclusion of a process identifier must apply to all keys in that \refstruct{pmix_query_t}. Queries for information on multiple specific processes therefore requires submitting multiple \refstruct{pmix_query_t} structures, each referencing one process.  Directives which are not applicable to a key are ignored. 
 
+% Use of pmix_server_query_fn is covered in server interfaces chapter
 \reqattrend
 
 \optattrstart
```
Comment (dsolt) [inline Chap_API_Query.tex:224]: This was supposed to be removed as it was for the blocking API presentation.  These are not optional attributes, these are keys.  The blocking case is correct, see that one.

**Comment by dsolt** [comment]
[pmix-standard-query.pdf](https://github.com/pmix/pmix-standard/files/8017359/pmix-standard-query.pdf)

Here is a pdf showing the changes to this chapter with green highlighting for new text, red for changed text and grey for moved text.

File: Chap_API_Query.tex [hunk 2]
```diff
@@ -0,0 +1,662 @@
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+% Chapter: Query
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+\chapter{Query Operations}
+\label{chap:api_proc_mgmt:query}
+
+This chapter presents mechanisms for generalized queries that
+access information about the host environment and the system in general.
+The chapter presents the concept of a query followed by a detailed explanation 
+of the queries \acp{API} provided.  The chapter compares the use of these \acp{API} with \refapi{PMIx_Get}.  The chapter concludes with detailed information about how to use
```
Comment (abouteiller) [inline Chap_API_Query.tex:10]: queries -> query

File: Chap_API_Query.tex [hunk 3]
```diff
@@ -0,0 +1,662 @@
...
+
+\ac{PMIx} sometimes provides multiple methods by which an application can obtain information or services.  For this example, 
+note that one could alternatively use the \refapi{PMIx_Register_event_handler} \ac{API} to register for an event indicating incipient job termination, and then use the \refapi{PMIx_Job_control_nb} \ac{API} to request that the host \ac{SMS} generate an event a specified amount of time prior to reaching the maximum run time. 
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+\subsection{Query Structure}
+\declarestruct{pmix_query_t}
+
+A \ac{PMIx} query structure is composed of one or more keys and a list of qualifiers which provide additional information to describe the query.
+Keys which use the same qualifiers can be placed in the same query for compactness, though it is permissable to put each key in its own query.
```
Comment (abouteiller) [inline Chap_API_Query.tex:27]: permissable -> permissible

File: Chap_API_Query.tex [hunk 4]
```diff
@@ -0,0 +1,662 @@
...
+\pasteAttributeItem{PMIX_SERVER_ATTRIBUTES}
+\pasteAttributeItem{PMIX_HOST_ATTRIBUTES}
+\pasteAttributeItem{PMIX_TOOL_ATTRIBUTES}
+
+Note that inclusion of both the \refattr{PMIX_PROCID} directive and either the \refattr{PMIX_NSPACE} or the \refattr{PMIX_RANK} attribute will return a \refconst{PMIX_ERR_BAD_PARAM} result, and that the inclusion of a process identifier must apply to all keys in that \refstruct{pmix_query_t}. Queries for information on multiple specific processes therefore requires submitting multiple \refstruct{pmix_query_t} structures, each referencing one process.  Directives which are not applicable to a key are ignored. 
+
+% Use of pmix_server_query_fn is covered in server interfaces chapter
+\reqattrend
+
+An implementation is not required to support any keys.  If a key is unsupported, the implementation should handle that key in the same way that it is required to handle a key which it cannot find.
```
Comment (abouteiller) [inline Chap_API_Query.tex:113]: any +particular+ keys

File: Chap_API_Query.tex [hunk 5]
```diff
@@ -0,0 +1,662 @@
...
+\declareAttribute{PMIX_DAEMON_MEMORY}{"pmix.dmn.mem"}{float}{
+Megabytes of memory currently used by the \ac{RM} daemon on the node. OPTIONAL QUALIFERS: \refattr{PMIX_HOSTNAME} or \refattr{PMIX_NODEID} (defaults to caller's node).
+}
+%
+\declareAttribute{PMIX_CLIENT_AVG_MEMORY}{"pmix.cl.mem.avg"}{float}{
+Average Megabytes of memory used by client processes on node. OPTIONAL QUALIFERS: \refattr{PMIX_HOSTNAME} or \refattr{PMIX_NODEID} (defaults to caller's node).
+}
+
+\vspace{\baselineskip}
+The following attributes are used as qualifiers in queries regarding attribute support within the \ac{PMIx} implementation and/or the host environment:
```
Comment (abouteiller) [inline Chap_API_Query.tex:361]: This piece of text needs to be removed, it is replaced by the following opener in 8.1.5.

Revision exception.

File: Chap_API_Query.tex [hunk 6]
```diff
@@ -0,0 +1,662 @@
...
+\begin{arglist}
+\argin{m}{Pointer to the \refstruct{pmix_query_t} structure (handle)}
+\argin{n}{Number of qualifiers to be allocated (\code{size_t})}
+\end{arglist}
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+\section{\code{PMIx_Resolve_peers}}
+
+There are a number of common queries for which \ac{PMIx} provides convenience routines.
+These \acp{API} provide similified access to commonly requested queries.  Due to their simplified interface, these \acp{API} cannot be customized through the use of attributes.
```
Comment (abouteiller) [inline Chap_API_Query.tex:527]: simplified -> simplified

File: Chap_API_Query.tex (hunk 5 above)
Comment (dsolt) [reply Chap_API_Query.tex:361]: Upon further investigation, this text was meant to introduce a portion of the attributes starting with PMIX_CLIENT_FUNCTIONS.  When thing were moved around this was accidentally not moved with those.  I moved this back to where it was originally.

File: Chap_API_Query.tex [hunk 7]
```diff
@@ -0,0 +1,662 @@
...
+\pasteAttributeItem{PMIX_SERVER_ATTRIBUTES}
+\pasteAttributeItem{PMIX_HOST_ATTRIBUTES}
+\pasteAttributeItem{PMIX_TOOL_ATTRIBUTES}
+
+Note that inclusion of both the \refattr{PMIX_PROCID} directive and either the \refattr{PMIX_NSPACE} or the \refattr{PMIX_RANK} attribute will return a \refconst{PMIX_ERR_BAD_PARAM} result, and that the inclusion of a process identifier must apply to all keys in that \refstruct{pmix_query_t}. Queries for information on multiple specific processes therefore requires submitting multiple \refstruct{pmix_query_t} structures, each referencing one process.  Directives which are not applicable to a key are ignored. 
+
+% Use of pmix_server_query_fn is covered in server interfaces chapter
+\reqattrend
+
+An implementation is not required to support any keys.  If a key is unsupported, the implementation should handle that key in the same way that it is required to handle a key which it cannot find.
```
[the remark this answers is not in the document]
Comment (dsolt) [reply Chap_API_Query.tex]: done

File: Chap_API_Query.tex (hunk 3 above)
Comment (dsolt) [reply Chap_API_Query.tex]: done

File: Chap_API_Query.tex (hunk 2 above)
Comment (dsolt) [reply Chap_API_Query.tex]: done

**Comment by jjhursey** [comment]
PMIx ASC 1Q 2022

 * Passed the first vote with revision exception: 10 yes / 1 no / 0 abstain
 * Failed the first vote without the revision exception: 9 yes / 2 no / 0 abstain
 * This is eligible for a second vote in the 2Q 2022 ASC quarterly meeting

**Comment by jjhursey** [comment]
PMIx ASC 2Q 2022
 * Passed the second vote: 11 yes / 0 no / 0 abstain
 * Will be merged into the next release

**Comment by jjhursey** [comment]
Rebased version in https://github.com/pmix/pmix-standard/pull/407

**Comment by jjhursey** [comment]
Closing in favor of PR #407 - which is the same content, but rebased onto the current `master`

**Pull request closed without merging**